src/Button/index.js

```diff
@@ -0,0 +1,271 @@
+   100% { opacity: 0.5; }
+`
+
+// ============================================================================
+// Decorators
+// ============================================================================
+
+@withOIOContext
+@withResponsiveObjectProps([
+   'borderRadius',
+   'color',
+   'fontFamily',
+   'padding',
+   'size',
+   'sizeMultiplier',
+   'textColor',
+   'textSize',
+   'textTransform',
+   'textWeight',
+   'width'
+])
+
+@withDynamicResponsiveProps((props, breakpoint) => {
+   const {
+      OIOContext,
+      outline,
+      rounded
+   } = props
+
+   // Responsive Props
+   const borderRadius = props.borderRadius && props.borderRadius[breakpoint]
+   const color = props.color && props.color[breakpoint]
+   const fontFamily = props.fontFamily && props.fontFamily[breakpoint]
+   const padding = props.padding && props.padding[breakpoint]
+   const size = props.size && props.size[breakpoint]
+
+   const height = buttonSizeDefaults[size].height
+   const width = props.width[breakpoint]
+
+   const textColor = props.textColor && props.textColor[breakpoint]
+   const textSize = props.textSize && props.textSize[breakpoint]
+   const textTransform = props.textTransform && props.textTransform[breakpoint]
+   const textWeight = props.textWeight && props.textWeight[breakpoint]
+
+   const primaryButtonColor = color || OIOContext.highlightColor
+   const defaultButtonPadding = width === 'auto' ? `0px ${height}` : '0px'
+
+   const buttonStyle = {
+      borderRadius,
+      fontFamily,
+      height,
+      textColor,
+      textTransform,
+      textWeight,
+      width,
+      backgroundColor: primaryButtonColor,
+      border: 'none',
+      textSize: textSize || buttonSizeDefaults[size].textSize,
+      hoverBackgroundColor: tinycolor(primaryButtonColor).lighten(7).toString(),
+      hoverBorder: '',
+      minWidth: height,
+      padding: padding ? `0px ${padding}` : defaultButtonPadding
+   }
+
+   if (rounded) {
+      buttonStyle.borderRadius = `${parseInt(height, 10) / 2}px`
+   }
+
+   if (outline) {
+      buttonStyle.border = `2px solid ${tinycolor(primaryButtonColor).setAlpha(0.5).toString()}`
+      buttonStyle.backgroundColor = 'transparent'
+      buttonStyle.hoverBackgroundColor = 'transparent'
+      buttonStyle.hoverBorder = `2px solid ${tinycolor(primaryButtonColor).setAlpha(0.8).toString()}`
+      buttonStyle.textColor = primaryButtonColor
+   }
+
+   return buttonStyle
+})
+
+// ============================================================================
+// Component
+// ============================================================================
+
+export default class Button extends React.Component {
+   static propTypes = {
+      borderRadius: OIOResponsiveObjectPropType,
+      children: PropTypes.node,
+      className: PropTypes.string,
+      color: OIOResponsiveObjectPropType,
+      fontFamily: OIOResponsiveObjectPropType,
+      mode: PropTypes.oneOf(['disabled', 'normal', 'pulsing', 'loading']),
```

Comment:
I would add an optional `id` prop that can be accepted to all OIO components which I think is extremely useful for testing and debugging